mobilecoind/src/service.rs

```diff
@@ -1927,6 +1950,20 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
         }
         let local_block_index = num_blocks - 1;
 
+        // Get LastBlockInfo from our peers
+        let block_infos = self.get_last_block_infos();
+        // choose the block info which is latest in terms of block index (we may be
+        // isolated from some of the nodes)
+        let last_block_info =
+            if let Some(latest) = block_infos.into_iter().max_by_key(|info| info.block_index) {
+                latest
+            } else {
+                return Err(RpcStatus::with_message(
+                    RpcStatusCode::INTERNAL,
+                    "no peers reachable".to_owned(),
+                ));
+            };
```

Comment:
Optional/Equivalent: Use `Option::ok_or_else`

```suggestion
            block_infos.into_iter().max_by_key(|info| info.block_index).ok_or_else(
                || RpcStatus::with_message(
                    RpcStatusCode::INTERNAL,
                    "no peers reachable".to_owned(),
                ))?;
```